app/views/users/admin_login.html.erb

```diff
@@ -8,11 +8,27 @@
       <% if @error %><p><%= @error %></p><% end %>
 
       <% if @second_factor_required %>
-        <%=form_tag({}, method: :put) do %>
-          <%= label_tag(:second_factor_token, t('login.second_factor_description')) %>
-          <%= render 'common/second_factor_text_field' %><br><br>
-          <%= submit_tag t('submit')%>
-        <% end %>
+        <div id="primary-second-factor-form">
+          <%=form_tag({}, method: :put) do %>
+            <%= label_tag(:second_factor_token, t('login.second_factor_description')) %>
+            <%= render 'common/second_factor_text_field' %><br><br>
+            <%= submit_tag t('submit')%>
+          <% end %>
+        </div>
+
+         <%if @backup_codes_enabled%>
+
+          <div id="backup-second-factor-form" style="display: none">
+            <%= form_tag({}, method: :put) do%>
+              <%= label_tag(:second_factor_token, t("login.second_factor_backup_description")) %>
+              <%= render 'common/second_factor_backup_input' %><br><br>
+              <%= submit_tag(t("submit")) %>
+            <%end%>
+
+          </div>
+          <a href="#" id="toggle-form"><%=t "login.second_factor_backup" %></a>
```

Comment:
In general it's better to do `<a href>` instead of `<a href='#'>` - the first one styles the same way, but doesn't add a hash to the URL if the user clicks it.